lib/tapioca/generators/dsl.rb

```diff
@@ -0,0 +1,366 @@
+      end
+
+      sig { override.params(error: Error).void }
+      def handle_error(error)
+      end
+
+      sig do
+        params(
+          requested_constants: T::Array[String],
+          should_verify: T::Boolean,
+          quiet: T::Boolean,
+          verbose: T::Boolean
+        ).void
+      end
+      def build_dsl(requested_constants, should_verify: false, quiet: false, verbose: false)
+        load_application(eager_load: requested_constants.empty?)
+        abort_if_pending_migrations!
+        load_dsl_generators
+
+        if should_verify
+          say("Checking for out-of-date RBIs...")
+        else
+          say("Compiling DSL RBI files...")
+        end
+        say("")
+
+        outpath = should_verify ? Pathname.new(Dir.mktmpdir) : config.outpath
+        rbi_files_to_purge = existing_rbi_filenames(requested_constants)
+
+        compiler = Compilers::DslCompiler.new(
+          requested_constants: constantize(requested_constants),
+          requested_generators: constantize_generators(config.generators),
+          excluded_generators: constantize_generators(config.exclude_generators),
+          error_handler: ->(error) {
+            say_error(error, :bold, :red)
+          }
+        )
+
+        compiler.run do |constant, contents|
+          constant_name = T.must(Reflection.name_of(constant))
+
+          if verbose && !quiet
+            say("Processing: ", [:yellow])
+            say(constant_name)
+          end
+
+          filename = compile_dsl_rbi(
+            constant_name,
+            contents,
+            outpath: outpath,
+            quiet: should_verify || quiet && !verbose
+          )
+
+          if filename
+            rbi_files_to_purge.delete(filename)
+          end
+        end
+        say("")
+
+        if should_verify
+          perform_dsl_verification(outpath)
+        else
+          purge_stale_dsl_rbi_files(rbi_files_to_purge)
+
+          say("Done", :green)
+
+          say("All operations performed in working directory.", [:green, :bold])
+          say("Please review changes and commit them.", [:green, :bold])
+        end
+      end
+
+      sig { params(eager_load: T::Boolean).void }
+      def load_application(eager_load:)
+        say("Loading Rails application... ")
+
+        loader.load_rails_application(
+          environment_load: true,
+          eager_load: eager_load
+        )
+
+        say("Done", :green)
+      end
+
+      sig { void }
+      def abort_if_pending_migrations!
+        return unless File.exist?("config/application.rb")
+        return unless defined?(::Rake)
+
+        Rails.application.load_tasks
+        if Rake::Task.task_defined?("db:abort_if_pending_migrations")
+          Rake::Task["db:abort_if_pending_migrations"].invoke
+        end
+      end
+
+      sig { void }
+      def load_dsl_generators
+        say("Loading DSL generator classes... ")
+
+        Dir.glob([
+          "#{Tapioca::Config::DSL_COMPILERS_PATH}/*.rb",
```

Comment:
This proved to be a fun bug to hunt down. 

In `generator.rb` it was listed as 
```ruby
"#{dir}/compilers/dsl/*.rb"
```
which was silently failing in the background of all the integrations tests after migrating the logic to this file (we're now one level higher). After determining the cause, I opted to extract this into a `Config` constant.